### PR TITLE
FINERACT-942 Added and Enforced EmptyCatchBlock checkstyle

### PR DESCRIPTION
**Pull request by thesmallstar**
@vorburger I hope this is the correct approach to deal with empty catch blocks :)

 FINERACT-942

### CODE REVIEW

**Comment by vorburger** [comment]
> @vorburger I hope this is the correct approach to deal with empty catch blocks :)

Your job here is going to be a little bit more challenging, thus more interesting! :smiling_imp: 

What you are getting into here is kind of related to what I alluded to in FINERACT-932... there are 2 cases: You're finding a number of obvious plain wrong **empty** `catch` statements. For example, catching a `javax.mail.MessagingException` and then just ignoring and doing absolutely nothing about it is most certainly a clear bug! If your code is trying to send out an email, and that fails, probably your SMTP server is fried, or whatever - and you certainly do want an "operator" who supervises error logs (e.g. like I do on https://www.fineract.dev, using https://cloud.google.com/error-reporting) do know about that. E.g. that `GeneralSecurityException` you spotted most likely also fall into this category.

However, for e.g. `EmptyResultDataAccessException` I'm personally not entirely sure yet. What does that exception signal? Presumably Spring is indicating that some JDBC ResultSet was empty, when it should not have been? Is that OK, or a clear indication of a database corruption? That depends on the specific code... But if it is OK and can happen during "normal" execution, then... there is another problem - then some other Spring API which returns an empty list or an Optional or something like that should have been used instead of expecting an exception! Do you know what I mean?

BTW can you write up what I'm explaining here into a follow-up PR to #912 ? :smile:

**Comment by vorburger** [comment]
> BTW can you write up what I'm explaining here into a follow-up PR to #912 ?

FYI I kind of just did that myself - hope you don't mind. Let's get #912 merged, and you can then improve upon it in this or future PRs, if anything is new or missing.

**Comment by thesmallstar** [comment]
@vorburger I am looking into what you said, will update shortly on this :)

**Comment by thesmallstar** [comment]
@vorburger I will continue of this now :D
Thank you for the detailed view on how to work on this.
From what I understand, I will go with two approaches firstly if it is "really an exception" and needs to inform to the operator I will log sufficient description of the error. 
Otherwise, If this might indicate a "bad approach to solve a problem" and might need a complete restructure of the code, I will try to do it :)

In both cases, I think looking what exactly an exception means is important in this case?  and is the solution :)

**Comment by vorburger** [comment]
> @vorburger I will continue of this now :D

@thesmallstar don't forget about this one, it would be great to see this wrapped up... this week?

**Comment by thesmallstar** [comment]
Yes, already working on this would be done in a day or two max :)

**Comment by thesmallstar** [comment]
@vorburger I wrote how I approached most of the exceptions in comment.
I did not see a major problem(That like what you suggested) please suggest if you have anything on the comments :) 
If you don't have anything to add, we can just remove the comment and merge this.

EDIT: please ignore that I removed the check of line endings, will be fixed before me merge this after review of comments.

**Comment by thesmallstar** [comment]
Another mistake in comments I have explained that:
EmptyResultDataAccessException is thrown when more than one row is returned by the query. It is actually thrown when atleast one row is expected and zero received
(Sorry for that)
The explanation for the count (*) still remains the same.
It is bound to return a single row

File: fineract-provider/config/checkstyle/checkstyle.xml
Comment (vorburger) [inline fineract-provider/config/checkstyle/checkstyle.xml]: why comment this out here for making this change?

File: fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountWritePlatformServiceJpaRepositoryImpl.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountWritePlatformServiceJpaRepositoryImpl.java]: :+1:

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/sms/service/SmsCampaignDropdownReadPlatformServiceImpl.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/sms/service/SmsCampaignDropdownReadPlatformServiceImpl.java]: you could already change this now? (or later, up to you; later is perfectly fine with me too)

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/config/checkstyle/checkstyle.xml
Comment (thesmallstar) [reply fineract-provider/config/checkstyle/checkstyle.xml]: Oh I think you missed it, I wrote a note above already that this is unintentional and will be fixed when I remove the comments

**Comment by thesmallstar** [comment]
Will update this now :) 

Update: Faced some minor issues, still working on this should be done by tomorrow.

**Comment by thesmallstar** [comment]
Update:
1. I have gone through all the empty catch in detail, I found no catch block that was suspicious, in most of the cases they were left empty because we were in a special case where we won't enter the catch block.
2. My past observations are on this branch if anyone wants to refer: https://github.com/thesmallstar/fineract/tree/refer
Note that, EmptyResultDataAccessException is thrown when more than one row is returned by the query. It is actually thrown when atleast one row is expected and zero received
(I wrote it incorrectly in comments, Sorry for that)
The explanation for the count (*) still remains the same.
It is bound to return a single row
3. Use of Exception class is not yet done, it will be the very next checkstyle I work on. I have not done it here to collectively address them all in a single PR.

**Review by vorburger — APPROVED** [review]
LGTM

**Pull request merged**
